File: Digitize.go

//https://www.codewars.com/kata/5583090cbe83f4fd8c000051

package main

import "fmt"

func Digitize(n int) []int {
  // your code here
  if n>0{
      var arr []int
  for n>0{
    temp:=n%10
    arr=append(arr,temp)
    n=n/10
  }
  return arr
  }
  return []int{0}
}

func main(){

fmt.Println(Digitize(35231))

}

File: digitize.cpp

//https://www.codewars.com/kata/5417423f9e2e6c2f040002ae

#include <iostream>
#include <vector>

using namespace std;

std::vector<int> digitize(int n) {
  // your code here
  std::vector<int> a;
  if (n==0){
    a.insert(a.begin(),n);
    return a;
  }
  
  while(n>0){
    int t=n%10;
    a.insert(a.begin(),t);
    n /=10;
    
  }
  return a;
}

int main(){
	
	vector<int> n=digitize(123);
	
	for (int i=0; i<n.size(); i++){
		cout<<n[i]<<" ";
	}
	
}
